### ISSUE TITLE
TypeError: <class 'datetime.date'> type object 2015-03-27

### ISSUE THREAD

**Issue by yoonguk510** [WillKoehrsen/Data-Analysis#16]
I am getting this error

  File "D:\anaconda\lib\site-packages\pandas\core\indexes\datetimes.py", line 133, in wrapper

TypeError: <class 'datetime.date'> type object 2015-03-27

**Comment by julietxiao** [comment]
I also met this error... Have you solve this problem?

**Comment by yoonguk510** [comment]
Yes I have. There is a problem with the python file. You have to run it
via jupiter notebook. It will work fine.

**Comment by WillKoehrsen** [comment]
I think this is an issue with the version of Pandas. Which version of Pandas were you using when you got the error?

**Comment by VictorHM** [comment]
> I think this is an issue with the version of Pandas. Which version of Pandas were you using when you got the error?

I think you are spot on. I got this message regarding this:

>  FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
> 'datetime.date' is coerced to a datetime. In the future pandas will
> not coerce, and a TypeError will be raised. To retain the current
> behavior, convert the 'datetime.date' to a datetime with
> 'pd.Timestamp'.

In the installation I have for Stocker, this is the version of Pandas and its dependencies Note that it is in a conda env that makes everything work (I had troubles making fbprophet work due to versions of pystan and others):

```
>>> pd.show_versions(as_json=False)

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.6.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.4
pytest: None
pip: 18.1
setuptools: 40.6.3
Cython: 0.29.3
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```

Hope this helps. Will try to go into it to see if I can modify Stocker.py to use new Pandas method. Regards.

**Comment by sebdau** [comment]
I'm having the same problem and went through similar install issues as described by @VictorHM.

I'm new to Python so I'd apppriciate if one could point out a solution or workaround.

Thanks a lot!

**Comment by sebdau** [comment]
#22 Solved the type problem for me!

**Comment by htuhxf** [comment]
I got the same problem with pandas0.23.4, just as described by VictorHM.
Solved the problem by "pd.to_datetime(df['date'], format='%Y%m%d').apply(lambda x: x.date())".[the stackoverflow link is here](https://stackoverflow.com/questions/41783003/how-do-i-convert-timestamp-to-datetime-date-in-pandas-dataframe)

**Comment by yoonguk510** [comment]
Late, but thank you for your help :)

**Comment by angelotc** [comment]
wrapping the datetime in `pd.Timestamp(<your_date>)` worked for me.

**Comment by angelotc** [comment]
> I think you are spot on. I got this message regarding this:
> 
> > FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
> > 'datetime.date' is coerced to a datetime. In the future pandas will
> > not coerce, and a TypeError will be raised. To retain the current
> > behavior, convert the 'datetime.date' to a datetime with
> > 'pd.Timestamp'.

wrapping the datetime in `pd.Timestamp(<your_date>)` worked for me.
